Python/ds/sumSubarrayMin.py

```diff
@@ -0,0 +1,47 @@
+def sum_subarray_mins(A) -> int:
+    """"returns the sum of the smallest elements of all arrays"""
+    stack = []
+    length = len(A)
+    left = [-1]*length
+    right = [length]*length
+    index = 0
+    while index < length:
+        while stack and A[stack[-1]] > A[index]:
+            right[stack.pop()] = index
+        else:
+            if stack:
+                left[index] = stack[-1]
+            
+        stack.append(index)
+        index += 1
+    sumi = 0
+    for index in range(length):
+        sumi += (index - left[index]) * (right[index]-index) * A[index]
+    return sumi
+
+arr = list(map(int, input().split(',')))
```

Comment:
It would be better if you prompt user how to enter elements  for better understanding.
eg: ```arr = list(map(int, input("Enter elements of an array (format- a,b,c..) : ").split(',')))```
and do format the output too. It should give some message with the answer.
eg: ```print("Sum of smallest elements of all sub-arrays is: {}" .format(sum_subarray_mins(arr)))```